jxline-core/src/main/java/cloud/xline/jxline/utils/Pair.java

```diff
@@ -0,0 +1,42 @@
+package cloud.xline.jxline.utils;
+
+import java.util.Objects;
+
+public final class Pair<A, B> {
+    final A a;
+    final B b;
+
+    Pair(A a, B b) {
+        this.a = Objects.requireNonNull(a);
+        this.b = Objects.requireNonNull(b);
+    }
+
+    /** Gets the first value. */
+    public A getFirst() {
+        return a;
+    }
+
+    /** Gets the second value. */
+    public B getSecond() {
+        return b;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o instanceof Pair<?, ?>) {
+            Pair<?, ?> e = (Pair<?, ?>) o;
+            return a.equals(e.getFirst()) && b.equals(e.getSecond());
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        return a.hashCode() ^ b.hashCode();
```

Comment:
This should be hash both a and b.

Reply:
It seems there is no such thing like `Hasher` in java.
`hashCode` is native method implemented in C/C++.

Reply:
Use `Objects.hash`? `a.hashCode() ^ b.hashCode()` will always result in a zero value if `a` equals `b`

Reply:
We will not create a type of `Pair(a, a)`, which is similar to Map.Entry.

https://docs.oracle.com/javase%2F7%2Fdocs%2Fapi%2F%2F/java/util/Map.Entry.html#hashCode()

Reply:
> Use `Objects.hash`?

A new object should not be created temporarily when calculating hash, as the resulting hash code would be random.